### PR TITLE
Fifth task commit. Waiting for review.

### PR DESCRIPTION
**Pull request by Radu-Stroe**
Scenario: Checking the edit functionality for my personal details  
   Given I launch the My Account URL
   And   I submit user credentials
   And   I click on Personal Details tab
   And   I successfully modify my Personal Details
   And   I am able to choose all the contact preferences available
   And   I click on Subscription Details tab
   When  I click on Personal Details tab
   Then  I can see that the chosen contact preferences remains stored

### CODE REVIEW

File: src/test/java/internship/framework/core/utility/TestData.java [hunk 1]
```diff
@@ -5,4 +5,11 @@
     private TestData() {
     }
 
+    //region yourDetails
+    public static final String YOUR_DETAILS_FIRST_NAME = "Radu";
```
Comment (cornel18i) [inline src/test/java/internship/framework/core/utility/TestData.java:6]: Why did we create a dedicated class just to store some constants? Are they related just for one test?

File: src/test/java/internship/framework/pages/PersonalDetailsPage.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    public PersonalDetailsPage fillWithValueCityTown() {
+        ProjectLogger.info("Fill city/town element with value.");
+        ElementUtils.fillInWithValue(cityTownElement, TestData.YOUR_DETAILS_CITY_TOWN);
+        return this;
+    }
+
+    public PersonalDetailsPage selectCountry() {
+        ProjectLogger.info("Select country.");
+        Select select = new Select(countryElement);
+        select.selectByVisibleText("Romania");
```
Comment (cornel18i) [inline src/test/java/internship/framework/pages/PersonalDetailsPage.java]: Why is "Romania" not a constant ?

File: src/test/java/internship/framework/pages/PersonalDetailsPage.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    }
+
+    public void navigatePersonalDetailsTab() {
+        ProjectLogger.info("Navigate to Personal Details page.");
+        ElementUtils.clickOn(personalDetailsButton);
+    }
+
+    private boolean isFirstCheckBoxSelected() {
+        ProjectLogger.info("Check if first contact checkbox is selected.");
+        Driver.get().manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
```
Comment (cornel18i) [inline src/test/java/internship/framework/pages/PersonalDetailsPage.java]: Why do we do this only for the first checkbox ?

File: src/test/java/internship/framework/pages/PersonalDetailsPage.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        return Boolean.parseBoolean(checkThirdCheckboxElement.getAttribute("checked"));
+    }
+
+    private boolean isFourthCheckBoxSelected() {
+        ProjectLogger.info("Check if fourth contact checkbox is selected.");
+        return Boolean.parseBoolean(checkFourthCheckboxElement.getAttribute("checked"));
+    }
+
+    public boolean areALLCheckboxesSelected() {
+        return isFirstCheckBoxSelected() & isSecondCheckBoxSelected()
```
Comment (cornel18i) [inline src/test/java/internship/framework/pages/PersonalDetailsPage.java]: Why did you use & instead of && ? Please read the following article : https://www.dummies.com/programming/java/and-operators-and-in-java/

File: src/test/java/internship/tests/PersonalDetailsTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+public class PersonalDetailsTest extends TestBase {
+    private ThreadLocal<SoftAssert> softAssert = new ThreadLocal<>();
+
+    @Test(testName = "Personal details test",
+            description = "Testing personal details.")
+    public void checkPersonalDetailsFunctionality() {
+        softAssert.set(new SoftAssert());
```
Comment (cornel18i) [inline src/test/java/internship/tests/PersonalDetailsTest.java]: Can we move line 17,18 from the test method somewhere else? This doesn't look related for the test.

File: src/test/java/internship/tests/PersonalDetailsTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    @AfterMethod(alwaysRun = true)
+    public void tearDown() {
+        softAssert.remove();
+        super.tearDown();
+    }
+
+    private void login()
```
Comment (cornel18i) [inline src/test/java/internship/tests/PersonalDetailsTest.java]: Can we move this method in a test utility class? Most likely you are going to need it for other tests. Also, it will be good if you can allow to manually set the username and password value. This way you are not limited to only one case.

**Review by cornel18i — CHANGES_REQUESTED** [review]

**Comment by cornel18i** [comment]
Also have a look at : https://app.codacy.com/app/cristina.nedin/internship2019/pullRequest?prid=3977707 . Codacy found 5 issues.

File: src/test/java/internship/framework/core/utility/TestData.java (hunk 1 above)
Comment (Radu-Stroe) [reply src/test/java/internship/framework/core/utility/TestData.java:6]: TestData class was already part of Cristina's testing framework. She advised me to use that class for these constants.

File: src/test/java/internship/framework/pages/PersonalDetailsPage.java (hunk 2 above)
Comment (Radu-Stroe) [reply src/test/java/internship/framework/pages/PersonalDetailsPage.java]: Noted and fixed.

File: src/test/java/internship/framework/pages/PersonalDetailsPage.java (hunk 3 above)
Comment (Radu-Stroe) [reply src/test/java/internship/framework/pages/PersonalDetailsPage.java]: I need the implicit wait time between navigating to Personal details page and checking the first checkbox. I moved that line inside test method.

File: src/test/java/internship/framework/pages/PersonalDetailsPage.java (hunk 4 above)
Comment (Radu-Stroe) [reply src/test/java/internship/framework/pages/PersonalDetailsPage.java]: My initial code include &&, but I got an advice from IntelliJ to simplify code this way. I've read the article and understand that is beneficial to use && performance-wise.

File: src/test/java/internship/tests/PersonalDetailsTest.java (hunk 5 above)
Comment (Radu-Stroe) [reply src/test/java/internship/tests/PersonalDetailsTest.java]: Line 17 . Cristina's advice was to set a new SoftAssert object at the begining of every test and remove it after every testing, using tearDown method.
Line 18. login() method is part of this test requirement.

File: src/test/java/internship/tests/PersonalDetailsTest.java (hunk 6 above)
Comment (Radu-Stroe) [reply src/test/java/internship/tests/PersonalDetailsTest.java]: Noted. I've created a new class TestUtils and moved the login() method there. Changed the method to allow manually set the username and password.

File: src/test/java/internship/framework/core/utility/TestUtils.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+public class TestUtils {
+
+    private TestUtils() {
+    }
+
+    public static void login() {
+        LoginPage loginPage = new LoginPage().openPage();
+        loginPage.clickLogInButtonHome();
+        loginPage.enterEmail("stroeradu03@gmail.com")
```
Comment (cornel18i) [inline src/test/java/internship/framework/core/utility/TestUtils.java]: The method should have the option to set the email(username) and password. By doing so you are not constraint with a single case. e.g: test scenarios with different users.

File: src/test/java/internship/tests/PersonalDetailsTest.java (hunk 6 above)
Comment (cornel18i) [reply src/test/java/internship/tests/PersonalDetailsTest.java]: I think you forgot the last part.

File: src/test/java/internship/tests/PersonalDetailsTest.java (hunk 5 above)
Comment (cornel18i) [reply src/test/java/internship/tests/PersonalDetailsTest.java]: **Line 17 . Cristina's advice was to set a new SoftAssert object at the begining of every test and remove it after every testing, using tearDown method.** So, how can we do this for line 17? Instead of copy/pasting the same line for each test?

Comment (cornel18i) [reply src/test/java/internship/tests/PersonalDetailsTest.java]: Line 18 can remain. If its part only for this test.

File: src/test/java/internship/tests/PersonalDetailsTest.java (hunk 6 above)
Comment (Radu-Stroe) [reply src/test/java/internship/tests/PersonalDetailsTest.java]: Fixed.

File: src/test/java/internship/framework/core/utility/TestUtils.java (hunk 7 above)
Comment (Radu-Stroe) [reply src/test/java/internship/framework/core/utility/TestUtils.java]: Fixed.

**Review by cornel18i — APPROVED** [review]

**Review by andreibeta — APPROVED** [review]
